src/Psalm/Context.php

```diff
@@ -477,7 +477,7 @@ public static function filterClauses(
             $quoted_remove_var_id = preg_quote($remove_var_id);
 
             foreach ($clause->possibilities as $var_id => $_) {
-                if (preg_match('/' . $quoted_remove_var_id . '[\]\[\-]/', $var_id)) {
+                if (preg_match('/' . preg_quote($quoted_remove_var_id, '/') . '[\]\[\-]/', $var_id)) {
```

Comment:
I was seeing `Warning: preg_match(): Unknown modifier 'a' in /path/to/project/vendor/vimeo/psalm/src/Psalm/Context.php on line 480` here

Reply:
I undid this change somewhat in 2100166, as it caused bugs

Reply:
The new change makes more sense, I was looking at the variable name but not the definition